app/charts/map/map-state.tsx

```diff
@@ -145,21 +144,13 @@ const useMapState = ({
 
   const getLabel = useStringVariable(fields.areaLayer.componentIri);
   const getValue = useOptionalNumericVariable(fields.areaLayer.measureIri);
-  const getFeatureLabel = useCallback(
-    (d: Observation | undefined): string =>
-      d ? `${d[fields.areaLayer.label.componentIri]}` : "",
-    [fields.areaLayer.label.componentIri]
-  );
   const getRadius = useOptionalNumericVariable(fields.symbolLayer.componentIri);
 
   const areaMeasureLabel =
-    measures
-      .find((m) => m.iri === fields["areaLayer"].measureIri)
-      ?.label.split("_")[1] || "";
+    measures.find((m) => m.iri === fields["areaLayer"].measureIri)?.label || "";
   const symbolMeasureLabel =
-    measures
-      .find((m) => m.iri === fields["symbolLayer"].componentIri)
-      ?.label.split("_")[1] || "";
+    measures.find((m) => m.iri === fields["symbolLayer"].componentIri)?.label ||
+    "";
```

Comment:
It looks like those calls could be either memoized or used an index, I am always a bit worried when I see a list scan in a render call.